src/app/builders/components/Table/BuildersTable.tsx

```diff
@@ -0,0 +1,216 @@
+'use client'
+
+import { AllocationsContext } from '@/app/collective-rewards/allocations/context'
+import { Token, useGetBuildersRewards } from '@/app/collective-rewards/rewards' // FIXME: change path so as to not import from a cousin folder
+import { Builder } from '@/app/collective-rewards/types'
+import {
+  isBuilderActive,
+  isBuilderDeactivated,
+  isBuilderKycRevoked,
+  isBuilderPaused,
+} from '@/app/collective-rewards/utils'
+import TablePager from '@/components/TableNew/TablePager'
+import { getTokens } from '@/lib/tokens'
+import {
+  Row,
+  SelectedRows,
+  usePricesContext,
+  useTableActionsContext,
+  useTableContext,
+  withTableContext,
+} from '@/shared/context'
+import { useReadGauges } from '@/shared/hooks/contracts'
+import { Suspense, useContext, useEffect, useMemo, useState } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+import BuilderFilterDropdown, { BuilderFilterOptionId } from '../../BuilderFilterDropdown'
+import { BuilderDataRow, convertDataToRowData } from './BuilderDataRow'
+import { BuilderHeaderRow } from './BuilderHeaderRow'
+import { ColumnId, DEFAULT_HEADERS, PAGE_SIZE } from './BuilderTable.config'
+import { Action } from './Cell/ActionCell'
+
+// --- Filter builders by state ---
+const filterInactive = (builder: Builder) => !isBuilderActive(builder.stateFlags)
```

Comment:
Maybe we need to add a TODO, to do the proper validation about the builders states.